test/services/invoice.service.test.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const { DatabaseHelper, GeneralHelper } = require('../support/helpers')
+
+// Thing under test
+const { InvoiceService } = require('../../app/services')
+
+describe('Invoice service', () => {
+  let transaction
+
+  const dummyTransaction = {
+    billRunId: 'f0d3b4dc-2cae-11eb-adc1-0242ac120002',
+    customerReference: 'CUSTOMER_REFERENCE',
+    chargeFinancialYear: 2021,
+    chargeCredit: false,
+    chargeValue: 5678
+  }
+
+  beforeEach(async () => {
+    await DatabaseHelper.clean()
+
+    // We clone the request fixture as our payload so we have it available for modification in the invalid tests. For
+    // the valid tests we can use it straight as
+    transaction = GeneralHelper.cloneObject(dummyTransaction)
+  })
+
+  describe('When a valid transaction is supplied', () => {
```

Comment:
These are great. I think I'd also like to see a couple of tests that cover

- We create a new `invoice` if one does not exist
- We get and update an existing `invoice` if a matching one already exists

Shout if you need any help with those.